bpf/lib/common.h

```diff
@@ -188,6 +188,20 @@ struct policy_entry {
 	__u64		bytes;
 };
 
+struct data_metrics_key {
+    __u8      reason;     //0: forwarded, >0 dropped
```

Comment:
There should probably be an enum for the 'reason' as well, METRIC_REASON_FORWARDED, METRIC_REASON_DROPPED, ... to support more metrics in the future.

Also just a question, would it be easier to encode the direction in  the reason rather than have a dir bit? So
enum {
  METRIC_REASON_FORWARD_INGRESS,
  METRIC_REASON_FORWARD_EGRESS,
  METRIC_REASON_DROP_INGRESS,
  METRIC_REASON_DROP_EGRESS,
};

In the future we could extend this to be even more specific if needed, METRIC_REASON_DROP_SOCKET for example comes to mind.  I'm not entirely up to speed on the prometheous reqs so maybe the bit has some meaning for mapping into higher layers?

Reply:
OK seems reasonable, thanks for the explanation. The '-reason' is a bit ugly but I don't see any good way to fix that easily and its a really big nit on my part.